addons/web/controllers/main.py

```diff
@@ -1884,11 +1884,20 @@ def content_type(self):
         """ Provides the format's content type """
         raise NotImplementedError()
 
+    @property
+    def extension(self):
+        raise NotImplementedError()
+
     def filename(self, base):
         """ Creates a valid filename for the format (with extension) from the
          provided base name (exension-less)
         """
-        raise NotImplementedError()
+        if base not in request.env:
+            return osutil.safe_filename(base) + self.extension
+
+        model_desc = request.env['ir.model']._get(base).name
+        filename = _("%(screen_title)s(%(base)s)", screen_title=model_desc, base=base)
```

Comment:
I'd be surprised if translators know what to do with `"%(screen_title)s(%(base)s)"`, that's just gibberish to them, especially as they don't know this to be a file name. Besides, `screen_title` is not really a screen title and `base` isn't ... what's a `base`?

Reply:
Good point. I updated these to "model description" and "model name", hopefully that's a bit clearer.

For the Pivot and Cohort views I used "title" and "model_name" but it's... debatable: the second should be the model name, if the first is missing then the model description is used but the views send the `@string` attribute, which *is not shown anywhere in the interface* and tends to be... not great e.g. for the opportunities pipeline, the action is called "Pipeline" (which is what appears on the screen) but the view just has `@string="Opportunities"` so the title is just "Opportunities", which is... meh.

From the task feedback it doesn't seem like ngr cares too much, and I guess that can always be changed later on.

Reply:
> Good point. I updated these to "model description" and "model name", hopefully that's a bit clearer.

Actually if you have no English content and only placeholders, you don't need to translate it at all. It could be useful if we need to change the order of the words but I don't see why it would be useful here.

Reply:
@mart-e sure but the main issue is with the cohort and pivot views which do have content.